### PR TITLE
PR for migration client

### PR DESCRIPTION
**Pull request by Shakila**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: tools/migrator/pom.xml [hunk 1]
```diff
@@ -0,0 +1,186 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
```
Comment (Nirothipan) [inline tools/migrator/pom.xml]: Shall we add the licence header?

File: tools/migrator/src/main/java/org/wso2/carbon/ei/migration/MigrationClient.java [hunk 2]
```diff
@@ -0,0 +1,6 @@
+package org.wso2.carbon.ei.migration;
```
Comment (vinok88) [inline tools/migrator/src/main/java/org/wso2/carbon/ei/migration/MigrationClient.java]: Add Licence headers and doc comments.

File: tools/migrator/src/main/java/org/wso2/carbon/ei/migration/MigrationClientException.java [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package org.wso2.carbon.ei.migration;
+
+public class MigrationClientException extends Exception {
```
Comment (vinok88) [inline tools/migrator/src/main/java/org/wso2/carbon/ei/migration/MigrationClientException.java]: Add Licence headers and doc comments.

File: tools/migrator/src/main/java/org/wso2/carbon/ei/migration/MigrationClientImpl.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+ */
+public class MigrationClientImpl implements MigrationClient {
+
+    private static final Log log = LogFactory.getLog(MigrationClientImpl.class);
+
+    @Override
+    public void execute() throws MigrationClientException {
+
+        try {
+            log.info("............................................................................................");
```
Comment (vinok88) [inline tools/migrator/src/main/java/org/wso2/carbon/ei/migration/MigrationClientImpl.java]: You can use a simple loop here.

File: tools/migrator/src/main/java/org/wso2/carbon/ei/migration/MigrationHolder.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+import java.util.List;
+
+/**
+ * Holder class to hold version migrator objects.
+ */
+public class MigrationHolder {
+
+    private static MigrationHolder migrationHolder = new MigrationHolder();
+    private List<Migrator> migrationList = new ArrayList<>();
+
```
Comment (vinok88) [inline tools/migrator/src/main/java/org/wso2/carbon/ei/migration/MigrationHolder.java]: Remove unnecessary empty lines.

File: tools/migrator/src/main/java/org/wso2/carbon/ei/migration/service/dao/DataSourceDAO.java [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+        try {
+            if (dsCollection != null) {
+                String[] dataSourceNames = dsCollection.getChildren();
+                for (String dataSourcePath : dataSourceNames) {
+                    Resource resource = registry.get(dataSourcePath);
+                    dataSources.add(resource);
+                }
+            }
+        } catch (Exception e) {
+            throw new MigrationClientException("Error in getting all data source names from repository: " + e.getMessage());
```
Comment (vinok88) [inline tools/migrator/src/main/java/org/wso2/carbon/ei/migration/service/dao/DataSourceDAO.java]: Look like the line length has exceeded. Shall we format?

File: tools/migrator/src/main/java/org/wso2/carbon/ei/migration/service/dao/DataSourceDAO.java [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+        try {
+            registry = getRegistry(tenantId);
+        } catch (DataSourceException e) {
+            throw new MigrationClientException("Error in getting the registry configuration: " + e.getMessage());
+        }
+        Collection dsCollection = null;
+        try {
+            dsCollection = (Collection) registry.get(DataSourceConstants.DATASOURCES_REPOSITORY_BASE_PATH);
+        } catch (RegistryException e) {
+            //Ignore
```
Comment (vinok88) [inline tools/migrator/src/main/java/org/wso2/carbon/ei/migration/service/dao/DataSourceDAO.java]: Why are we ignoring this?

File: tools/migrator/src/main/java/org/wso2/carbon/ei/migration/service/dao/DataSourceDAO.java [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+        }
+        try {
+            if (dsCollection != null) {
```
Comment (vinok88) [inline tools/migrator/src/main/java/org/wso2/carbon/ei/migration/service/dao/DataSourceDAO.java]: What happens if the dsCollection == null ?

File: tools/migrator/src/main/java/org/wso2/carbon/ei/migration/service/dao/EventSinkDAO.java [hunk 9]
```diff
@@ -0,0 +1,189 @@
...
+        EventSink eventSink = new EventSink();
+        String filePath = EventSinkXmlReader.getTenantDeployementDirectoryPath();
+        File eventSinkFile = new File(filePath + name);
+        if (eventSinkFile.exists()) {
+            eventSink.setName(eventSinkFile.getName());
+            FileInputStream fileInputStream = null;
+
+            try {
+                fileInputStream = new FileInputStream(eventSinkFile);
+                eventSink = eventSinkConfigBuilder.createEventSinkConfig(Utility.toOM(fileInputStream), FilenameUtils.removeExtension(eventSink.getName()));
```
Comment (vinok88) [inline tools/migrator/src/main/java/org/wso2/carbon/ei/migration/service/dao/EventSinkDAO.java]: Reformat here and all other places that have exceeded the standard line length.

File: tools/migrator/src/main/java/org/wso2/carbon/ei/migration/service/dao/ServerProfileDAO.java [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+                }
+            }
+
+            if (newEncryptedPassword != null) {
+                OutputStream outputStream = new FileOutputStream(new File(filePath));
+                documentElement.serialize(outputStream);
+                isModified = true;
+            }
+        } catch (XMLStreamException | FileNotFoundException e) {
+            new MigrationClientException("Error while writing the file: " + e);
```
Comment (vinok88) [inline tools/migrator/src/main/java/org/wso2/carbon/ei/migration/service/dao/ServerProfileDAO.java]: don't we need to throw this exception?

File: tools/migrator/src/main/java/org/wso2/carbon/ei/migration/service/dao/ServerProfileDAO.java [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+        } catch (CryptoException e) {
+            e.printStackTrace();
```
Comment (vinok88) [inline tools/migrator/src/main/java/org/wso2/carbon/ei/migration/service/dao/ServerProfileDAO.java]: Shall we log this instead?

File: tools/migrator/src/main/java/org/wso2/carbon/ei/migration/service/migrator/EntitlementMediatorMigrator.java [hunk 12]
```diff
@@ -0,0 +1,175 @@
...
+        try {
+            tenants = MigrationServiceDataHolder.getRealmService().getTenantManager().getAllTenants();
+            boolean isIgnoreForInactiveTenants = Boolean.parseBoolean(System.getProperty(Constant.IGNORE_INACTIVE_TENANTS));
+            for (Tenant tenant : tenants) {
+                if (isIgnoreForInactiveTenants && !tenant.isActive()) {
+                    log.info("Tenant " + tenant.getDomain() + " is inactive. Skipping secondary userstore migration!");
+                    continue;
+                }
+
+                HashMap<String, File[]> filesMap = EntitlementMediatorDAO.getInstance().getEMConfigFiles(tenant.getId());
```
Comment (vinok88) [inline tools/migrator/src/main/java/org/wso2/carbon/ei/migration/service/migrator/EntitlementMediatorMigrator.java]: Reformat the line.

File: tools/migrator/src/main/java/org/wso2/carbon/ei/migration/util/Constant.java [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+*/
+package org.wso2.carbon.ei.migration.util;
+
+import javax.xml.namespace.QName;
+
+/**
+ * Holds common constants in migration service.
+ */
+public class Constant {
+    public static final String RSA = "RSA" ;
```
Comment (prabushi) [inline tools/migrator/src/main/java/org/wso2/carbon/ei/migration/util/Constant.java]: Remove space before ";"

File: tools/migrator/src/main/java/org/wso2/carbon/ei/migration/util/Utility.java [hunk 14]
```diff
@@ -0,0 +1,204 @@
...
+* limitations under the License.
+*/
+package org.wso2.carbon.ei.migration.util;
+
+import org.apache.axiom.om.OMElement;
+import org.apache.axiom.om.impl.builder.StAXOMBuilder;
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+//import org.wso2.carbon.core.internal.CarbonCoreDataHolder;
```
Comment (prabushi) [inline tools/migrator/src/main/java/org/wso2/carbon/ei/migration/util/Utility.java]: Shall we remove commented lines?

File: tools/migrator/src/main/java/org/wso2/carbon/ei/migration/service/dao/DataSourceDAO.java (hunk 8 above)
Comment (Shakila) [reply tools/migrator/src/main/java/org/wso2/carbon/ei/migration/service/dao/DataSourceDAO.java]: If it's null, we don't need to do anything.

File: tools/migrator/src/main/java/org/wso2/carbon/ei/migration/service/dao/DataSourceDAO.java (hunk 7 above)
Comment (Shakila) [reply tools/migrator/src/main/java/org/wso2/carbon/ei/migration/service/dao/DataSourceDAO.java]: We have to continue if the repo path is not there.

File: tools/migrator/src/main/java/org/wso2/carbon/ei/migration/MigrationClientImpl.java [hunk 15]
```diff
@@ -0,0 +1,54 @@
...
+    private static final Log log = LogFactory.getLog(MigrationClientImpl.class);
+
+    @Override
+    public void execute() throws MigrationClientException {
+
+        try {
+            for (int i = 0; i < 5; i++) {
+                log.info("...........................................................................................");
+            }
+
```
Comment (Nirothipan) [inline tools/migrator/src/main/java/org/wso2/carbon/ei/migration/MigrationClientImpl.java]: Can we remove these empty lines?

File: tools/migrator/src/main/java/org/wso2/carbon/ei/migration/service/dao/EventSinkDAO.java [hunk 16]
```diff
@@ -0,0 +1,198 @@
...
+import java.io.FileWriter;
+import java.io.StringReader;
+import java.io.StringWriter;
+import java.io.Writer;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+
```
Comment (Nirothipan) [inline tools/migrator/src/main/java/org/wso2/carbon/ei/migration/service/dao/EventSinkDAO.java]: Can we add class comments?

File: tools/migrator/src/main/java/org/wso2/carbon/ei/migration/service/dao/ServerProfileDAO.java [hunk 17]
```diff
@@ -0,0 +1,103 @@
+package org.wso2.carbon.ei.migration.service.dao;
```
Comment (Nirothipan) [inline tools/migrator/src/main/java/org/wso2/carbon/ei/migration/service/dao/ServerProfileDAO.java]: Licence header missing

File: tools/migrator/src/main/java/org/wso2/carbon/ei/migration/service/migrator/JMXProfileDataMigrator.java [hunk 18]
```diff
@@ -0,0 +1,160 @@
+package org.wso2.carbon.ei.migration.service.migrator;
```
Comment (Nirothipan) [inline tools/migrator/src/main/java/org/wso2/carbon/ei/migration/service/migrator/JMXProfileDataMigrator.java]: Please add licence header in all files

File: tools/migrator/src/main/java/org/wso2/carbon/ei/migration/MigrationClientException.java (hunk 3 above)
Comment (Shakila) [reply tools/migrator/src/main/java/org/wso2/carbon/ei/migration/MigrationClientException.java]: I have added the licence headers and doc comments.

File: tools/migrator/src/main/java/org/wso2/carbon/ei/migration/MigrationClient.java (hunk 2 above)
Comment (Shakila) [reply tools/migrator/src/main/java/org/wso2/carbon/ei/migration/MigrationClient.java]: I have added the licence headers and doc comments.

**Pull request merged**